runtime/lib/component/ota.js

```diff
@@ -72,7 +73,7 @@ class OTA {
         if (!info) {
           return false
         }
-        this.openUrl(`yoda-skill://ota/on_first_boot_after_upgrade?changelog=${encodeURIComponent(info.changelog)}`)
+        this.runtime.openUrl(`yoda-skill://ota/on_first_boot_after_upgrade?changelog=${encodeURIComponent(info.changelog)}`)
```

Comment:
Shall we verify this if we write a typing for component?

Reply:
There is no force validation on `.js` file unless we translate all runtime codes to typescript.

Reply:
Also test coverage could uncover faults early too.

Reply:
Actually our codebase already in TypeScript, because that's the superset of JavaScript. What I meant is that:

1. Write a typings for `AppRuntime` and some components.
2. Provide a base class, which contains basic fields there, and also it owns a typing.

If a component extends from the above class, then the TypeScript validation should be working?

Reply:
That assuming the TypeScript check is working on .js files :)

Reply:
> Also test coverage could reveal faults early too.

Of course the test should resolve something, but we still figure out another way like introducing strong types partially to avoid faults, either.

Reply:
If `checkJs` and `allowJs` are both enabled on typescript configs, I believe typescript compiler would make type checks on JS files.

Reply:
Good! Anyway this LGTM, let's discussion this in another PR :)